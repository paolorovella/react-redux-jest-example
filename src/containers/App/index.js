import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from 'src/actions';
import { Button, Counter, Logo } from 'src/components';

import logoImg from 'src/assets/logo.svg';

import {
  ButtonsContainer,
  CounterContainer,
  Container,
  H2,
} from './styles';

const App = ({ counter, increment, decrement }) => (
  <Container>
    <Logo src={logoImg} />
    <H2>React Redux Jest example</H2>
    <ButtonsContainer>
      <Button type="success" onClick={increment}>Increment</Button>
      <Button type="danger" onClick={decrement}>Decrement</Button>
    </ButtonsContainer>
    <CounterContainer>
      <Counter value={counter} />
    </CounterContainer>
  </Container>
);

App.propTypes = {
  counter: PropTypes.number.isRequired,
  increment: PropTypes.func.isRequired,
  decrement: PropTypes.func.isRequired,
};

export default connect(
  state => ({ counter: state.counter.current }),
  dispatch => ({ ...bindActionCreators(actions, dispatch) }),
)(App);
