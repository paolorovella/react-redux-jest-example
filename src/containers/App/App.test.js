import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import store from 'src/store';
import App from '.';

it('<App /> renders without crashing', () => {
  const tree = renderer.create(
    <App store={store} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
