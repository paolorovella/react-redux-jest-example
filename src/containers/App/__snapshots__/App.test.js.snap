// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
.c4 {
  color: #2E7D32;
  border: 2px solid #2E7D32;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c4:hover {
  background-color: #1B5E20;
  color: white;
  border: 2px solid #1B5E20;
}

.c5 {
  color: #D32F2F;
  border: 2px solid #D32F2F;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c5:hover {
  background-color: #B71C1C;
  color: white;
  border: 2px solid #B71C1C;
}

.c1 {
  -webkit-animation: App-logo-spin infinite 20s linear;
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.c0 {
  max-width: 1440px;
  margin: auto;
  text-align: center;
  font-family: 'Open Sans', sans-serif;
}

.c2 {
  font-weight: 600;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin: auto;
  margin-top: 20px;
  max-width: 260px;
}

.c6 {
  margin-top: 20px;
}

<div
  className="c0"
>
  <img
    className="c1"
    src="logo.svg"
  />
  <h2
    className="c2"
  >
    React Redux Jest example
  </h2>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[Function]}
      type="success"
    >
      Increment
    </button>
    <button
      className="c5"
      onClick={[Function]}
      type="danger"
    >
      Decrement
    </button>
  </div>
  <div
    className="c6"
  >
    <span>
      Counter
      : 
      0
    </span>
  </div>
</div>
`;
